eternalcore-api/src/main/java/com/eternalcode/core/feature/randomteleport/RandomTeleportService.java

```diff
@@ -38,13 +38,27 @@ public interface RandomTeleportService {
     /**
      * Asynchronously retrieves a safe random location within the specified world, using radius.
      *
+     * @deprecated Use {@link #getSafeRandomLocation(World, RandomTeleportRadiusRepresenter, int)}
+     * instead with an appropriate RandomTeleportRadiusRepresenter implementation for your border shape.
+     *
      * @param world        The world in which to find a random location.
      * @param radius       The radius around the player to search for a safe location.
      * @param attemptCount The number of attempts to find a safe location.
      * @return A CompletableFuture containing the random Location that is deemed safe.
      */
+    @Deprecated(forRemoval = true)
     CompletableFuture<Location> getSafeRandomLocation(World world, int radius, int attemptCount);
 
+    /**
+     * Asynchronously retrieves a safe random location within the specified world, using radius.
+     *
+     * @param world        The world in which to find a random location.
+     * @param radius       The radius around the player to search for a safe location.
+     * @param attemptCount The number of attempts to find a safe location.
+     * @return A CompletableFuture containing the random Location that is deemed safe.
+     */
+    CompletableFuture<Location> getSafeRandomLocation(World world, RandomTeleportRadiusRepresenter radius, int attemptCount);
```

Comment:
Also, I want to change the name of this class and convert it to record because now this api is not easy to use